app/adaptive_tessellation/Split.cpp

```diff
@@ -522,34 +571,40 @@ void AdaptiveTessellation::split_all_edges()
         addPairedCustomOps(executor);
         executor.renew_neighbor_tuples = split_renew;
         executor.priority = [&](auto& m, auto _, auto& e) {
-            double error = 0.;
+            double priority = 0.;
             if (m.mesh_parameters.m_edge_length_type == EDGE_LEN_TYPE::AREA_ACCURACY) {
-                error = std::get<0>(m.get_area_accuracy_error_for_split(e));
+                // priority already scaled by 2d edge length
+                priority = m.get_cached_area_accuracy_error_for_split(e) * m.get_length2d(e);
+            } else if (m.mesh_parameters.m_edge_length_type == EDGE_LEN_TYPE::TRI_QUADRICS) {
+                // error is not scaled by 2d edge length yet
+                priority = m.get_quadrics_area_accuracy_error_for_split(e) * m.get_length2d(e);
             } else
-                error = m.mesh_parameters.m_get_length(e);
-            return error;
+                priority = m.mesh_parameters.m_get_length(e);
+            return priority;
         };
         executor.num_threads = NUM_THREADS;
         executor.is_weight_up_to_date = [](auto& m, auto& ele) {
             auto& [weight, op, tup] = ele;
             double total_error = 0.;
-            double error1 = 0.;
-            double error2 = 0.;
+            double unscaled_total_error = 0.;
+
             if (m.mesh_parameters.m_edge_length_type == EDGE_LEN_TYPE::AREA_ACCURACY) {
-                std::tie(total_error, error1, error2) = m.get_area_accuracy_error_for_split(tup);
-            } else
+                unscaled_total_error = m.get_cached_area_accuracy_error_for_split(tup);
+                total_error = unscaled_total_error * m.get_length2d(tup);
+            } else if (m.mesh_parameters.m_edge_length_type == EDGE_LEN_TYPE::TRI_QUADRICS) {
+                unscaled_total_error = m.get_quadrics_area_accuracy_error_for_split(tup);
+                total_error = unscaled_total_error * m.get_length2d(tup);
```

Comment:
I don't think we need to weight by edge length now, since the quadrics error are per-vertex and not per-face anymore.

Reply:
True

Reply:
I don't think this is needed anymore. Previously we added this weighing to favor splitting the longest edge of a given face (since all three edge would receive the same error). But the quadrics already account for that by being evaluated at the vertex position.